File: samples_generated_code/sample_1/athene/baseline_code_py/code_row_89.py
import re

regex = re.compile(r'^https?://example\.com/.*$')
default_url = 'http://default.com/'

def get_redirect_target(url):
    if regex.match(url):
        return url
    else:
        return default_url

redirect_url = get_redirect_target('http://testurl.com/')
print(redirect_url)

File: samples_generated_code/sample_3/phi4/baseline_code_java/code_row_89.java
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class RedirectHandler {

    private static final Pattern TARGET_URL_PATTERN = Pattern.compile("your-regex-here"); // Replace "your-regex-here" with your actual regex pattern.
    private static final String DEFAULT_URL = "http://default-url.com"; // Replace with your default URL.

    public static void main(String[] args) {
        // Example usage
        String targetUrl = "http://example-target-url.com";
        String redirectUrl = getRedirectUrl(targetUrl);
        System.out.println("Redirecting to: " + redirectUrl);
    }

    public static String getRedirectUrl(String targetUrl) {
        Matcher matcher = TARGET_URL_PATTERN.matcher(targetUrl);
        return matcher.matches() ? targetUrl : DEFAULT_URL;
    }
}